medusa/providers/generic_provider.py

```diff
@@ -94,7 +94,7 @@ def __init__(self, name):
         self.public = False
         self.search_fallback = False
         self.search_mode = None
-        self.session = make_session()
+        self.session = MedusaSession(hooks=[cloudflare])
```

Comment:
@labrys what if we subclass to MedusaSafeSession and add exception handling to it? Then we can safely transition to the session objects. And replace them with MedusaSession later, one by one.